src/Stratis.Bitcoin/FullNode.cs

```diff
@@ -92,6 +92,35 @@ public INodeLifetime NodeLifetime
         /// <inheritdoc />
         public IFullNodeServiceProvider Services { get; set; }
 
+        public T NodeController<T>(bool failWithDefault = false)
```

Comment:
Wouldn't it be easier to register controller with the DI instead of all this black magic?

Reply:
See below.

Reply:
This code seems to be easily moved to an extention method in the common test project.